pennylane/ops/qubit/observables.py

```diff
@@ -137,6 +137,10 @@ def compute_matrix(A: TensorLike) -> TensorLike:  # pylint: disable=arguments-di
         Hermitian._validate_input(A)
         return A
 
+    @staticmethod
+    def compute_sparse_matrix(A) -> csr_matrix:  # pylint: disable=arguments-differ
```

Comment:
? Why is this needed?

Reply:
Because when the interface is `numpy`, [this line](https://github.com/PennyLaneAI/pennylane/blob/5f4169a297605ab6c785eef11828aecb749067d5/pennylane/devices/qubit/measure.py#L199) returns `csr_dot_products`, which uses the sparse matrix.

Reply:
I might actually call this a bug in DQ.  We should only be using `csr_dot_products` if the observable actually provides a sparse matrix.

I'm wondering now if the performance will shift when we actually start using the optimized numpy logic instead of autograd logic for the simualtions.

Reply:
But this is potentially a sufficient fix for now.

Reply:
#6264